common/lib/transport/connectionmanager.js

```diff
@@ -123,6 +123,29 @@ var ConnectionManager = (function() {
 		/* intercept close event in browser to persist connection id if requested */
 		if(createCookie && options.recover === true && window.addEventListener)
 			window.addEventListener('beforeunload', this.persistConnection.bind(this));
+
+		/* Listen for online and offline events */
+		if(typeof window === "object" && window.addEventListener) {
+			var self = this;
+			window.addEventListener('online', function() {
+				if(self.state == states.disconnected || self.state == states.suspended) {
+					Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager caught browser ‘online’ event', 'reattempting connection');
+					self.requestState({state: 'connecting'});
+				}
+			});
+			window.addEventListener('offline', function() {
+				if(self.state == states.connected) {
+					Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager caught browser ‘offline’ event', 'disconnecting active transport');
+					// Not sufficient to just go to the 'disconnected' state, want to
+					// force all transports to reattempt the connection
+					self.disconnectAllTransports();
+					// Try a reconnect immediately rather than wait for the disconnect timeout
+					Utils.nextTick(function() {
```

Comment:
> I thought we would always go immediately into connecting when a disconnection occurs from connected

Doesn't seem to do that, no. Might be missing something, but I can't see any mechanism for that to happen - the retry timer in notifyState is set to whatever the retryDelay for the new state is, doesn't look at the previous state (except to check if it's already there or is a terminal state).